src/app/components/FacetSidebar/Facet.jsx

```diff
@@ -39,6 +39,10 @@ class Facet extends React.Component {
     }
   }
 
+  componentWillUnmount() {
+    clearTimeout(this.timeout);
```

Comment:
We have a function to clear timeout here to remove seTimeout in `componentDidMount`. It helps remove the warnings when run unit test with `-w` flag by avoiding setting state on not-yet-mounted component.